curp/src/cmd.rs

```diff
@@ -97,7 +97,7 @@ impl ConflictCheck for String {
     }
 }
 
-impl ConflictCheck for u32 {
+impl ConflictCheck for Vec<u8> {
```

Comment:
I can't get the point of why the engine cannot use an u32 integer as a key.

Reply:
Next PR will use `Engine` as `TestCE`'s store, and the `Engine` can only save `Vec<u8>` data,  Another possible solution is to do the conversion between u32 and Vec<u8> when data is written or read. what do you think

Reply:
Is affected by one engine implementation？Or is it a trait interface constration?

Reply:
`Curp`'s test directly uses `MemorySnapshot`, and it serializes the store by itself and writes it to the snapshot. This is not expected, so I will replace the store in the `Curp`'s test wit `MemoryEngine`. Our `MemoryEngine` can only read and write `Vec<u8>`, you can't use `u32` directly, so we need to fix this.
And I rewrote this PR by type conversion, now it looks easier